### PR TITLE
Warn if there are conflicts between exported functions and the globalenv

### LINKED ISSUE #112: Detect functions with the same name in .GlobalEnv
Consider the following scenario:

- You develop a package or an analysis project structured as a package, an RStudio project
- You use `load_all()` to load the functions
- You edit a function in a file in the `R/` directory
- Accidentally you source this file (Ctrl + Shift + Enter)
- Now the functions are defined in two places: `.GlobalEnv` and the package environment
- You edit the function
- `load_all()` doesn't apply the changes because the function is overridden in `.GlobalEnv`

I see several ways to approach this problem:

- Detect conflicts in `load_all()`: if `.GlobalEnv` contains functions by the same name as defined in the files in `R/`
- Augment the {conflicted} package to deal with this situation
- Make sure that the IDE never sources files in `R/`

Filing here to initiate the discussion.

### PR DESCRIPTION
**Pull request by jimhester**
Often when developing you will accidentally source a file or define a
function in the console which then masks the function exported by
`load_all()`, which can cause confusion as to why your changed in the
source file are not being used until you determine the issue.

load_all now issues an informative warning in these cases to make it
clear there is an issue and how to remedy it.

Fixes #112

### CODE REVIEW

File: R/load.r [hunk 1]
```diff
@@ -78,6 +78,11 @@
 #' @param recompile DEPRECATED. force a recompile of DLL from source code, if
 #'   present. This is equivalent to running [pkgbuild::clean_dll()] before
 #'   `load_all`
+#' @param warn_conflicts If `TRUE`, issue a warning if there are conflicts
+#'   between the exported functions and functions in the global namespace. This
+#'   most commonly happens when you accidently source a R file rather than using
```
Comment (jennybc) [inline R/load.r]:
```suggestion
#'   most commonly happens when you accidently source an R file rather than using
```

**Comment by krlmlr** [comment]
Neat! I have a few minor suggestions:

- 01314ce shortens and sorts the bullet list to at most three and points at the `R/` directory
- 07a2f38 uses a shortcut (early return) instead of a long `if`

Happy to update the PR if you agree.

**Comment by jimhester** [comment]
Thanks for the suggestions @krlmlr!

I added an early return and sorted the function names, thanks for the suggestion!

I don't think shortening is needed, part of the appeal of the bulleted list is it maintains readability even with many results, and realistically most of the time only a handful of functions will be masked.

I also don't think specifying the scripts are in `R/` is appropriate, this could just as easily happen when writing a vignette or even in the tests for instance, or just running an external script that is not part of the package.

**Comment by krlmlr** [comment]
With many results the bulleted list gets shortened, at least on my system. The `rm()` call still has all the info, but is hidden when too many bullets are shown.

**Comment by krlmlr** [comment]
#114 has the short bullet list on top of this branch, thanks for considering.

**Pull request merged**